modules/stochastic_tools/src/state_sim/model/events/StateSimEvStateCng.C

```diff
@@ -0,0 +1,86 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "StateSimEvStateCng.h"
+#include <string>
+
+StateSimEvStateCng::StateSimEvStateCng(std::string name)
+  : StateSimEvConditionBased(name, ET_STATE_CNG),
+    _if_in_state(true),
+    _all_items(true)
+{
+}
+
+StateSimEvStateCng::StateSimEvStateCng(std::string name, std::vector<StateSimState> & key_states, bool if_in_state, bool all_items)
+  : StateSimEvConditionBased(name, ET_STATE_CNG),
+    _if_in_state(if_in_state),
+    _all_items(all_items)
+{
+  for (auto state : key_states)
+  {
+    this->addRelatedItem(state);
+  }
+}
+
+bool
+StateSimEvStateCng::isTriggered(const TimespanH & cur_ev_time, const StateSimBitset & cur_states, const StateSimModel & main_model)
+{
+  UNUSED(cur_ev_time);
+  UNUSED(main_model);
+
+  bool found = false;
+  for (auto item : _related_items)
+  {
+    int i = item.second->id();
+    //const int & cnt = ;
+    if ((cur_states.count() > i) && (cur_states[i])) //desired state is in current list
+    {
+      found = true;
+
+      if (_if_in_state) //We are looking for an item in the list to trigger us
+      {
+        if (!_all_items) //only one item true needed.
+        {
+          return true;
+        }
+      }
+      else //don't want the item in order to trigger us but we found it
+      {
+        if (_all_items) //only one item needed to not trigger.
+        {
+          return false;
+        }
+      }
+    }
+    else //desired state is not in current list
+    {
+      if (_if_in_state) //We are looking for an item in the list to trigger us
+      {
+        if (_all_items) //must have all items for a tigger needed.
+        {
+          return false;
+        }
+      }
+      else //don't want the item in order to trigger and we didn't find it
+      {
+        if (!_all_items) //only one item needed trigger.
+        {
+          return true;
+        }
+      }
+    }
+  }
```

Comment:
We prefer to drop braces for single statements. It'll really collapse code like this down to fewer lines.